### DISCUSSION TITLE
push button not enabled

### DISCUSSION THREAD

**Discussion by RebeccaMayer** [IBAHCM/RPiR#561] [Q&A for technical issues] [answered]
Hi @richardreeve, I am having issues pushing from r to github. The button will not enable, is there a manual way to do this?
thanks!

**Comment by richardreeve** [comment]
The push button is most likely not enabled because you haven't followed the instructions to create the repository on github, as we did in [3.1](https://sbohvm.github.io/RPiR/tutorials/practical3-1/practical3-1.html), and is described more generally in the [R package reference guide we provided](https://sbohvm.github.io/RPiR/articles/pages/packages_guide.html):

```R
usethis::use_github(organisation = "SBOHVM", private = TRUE)
```

Once the R package is connected to github, the button should be enabled whenever there are commits to push.

**Reply by RebeccaMayer** [reply]
Hi @richardreeve, I have already tried using usethis::use_github(organisation = "SBOHVM", private = TRUE) but Error in `gh_process_response()`: GitHub API error (401): Bad credentials comes up

**Reply by mhuzzell** [reply]
Did your access key expire?

**Reply by RebeccaMayer** [reply]
I don't think so, I regenerated a new access token last week

**Reply by richardreeve** [reply]
Right, but once you've generated a new token you need to follow the instructions to use it in RStudio. You can see this [in an earlier question](https://github.com/SBOHVM/RPiR/discussions/551)

**Reply by RebeccaMayer** [reply]
I went through the instructions but it's still disabled

**Reply by richardreeve** [reply]
If you still have bad credentials, then you need to start again from scratch and create a new token:

```R
usethis::create_github_token()
```

and once you've created it, add it both to the `.Renviron` file (replacing the existing value there):

```R
usethis::edit_r_environ()
```

and changing the credentials here:

```R
gitcreds::gitcreds_set()
```

Otherwise you wouldn't have a bad credential in RStudio. That's how it works.

**Reply by richardreeve** [reply]
Weirdly you actually have two diversity repositories in SBOHVM @RebeccaMayer - https://github.com/SBOHVM/rebeccamayer1Diversity and https://github.com/SBOHVM/RebeccamayerDiversity - so at some point you've had this working, but it's impossible to tell remotely why you don't have anything working now.

**Reply by RebeccaMayer** [reply]
I've generated another token and used the instructions above but it's still the push button still won't enable

**Reply by RebeccaMayer** [reply]
I tried creating a new repository as I thought I had set it up incorrectly but nothing changed

**Reply by richardreeve** [reply]
Does it still give the 401 error? I've never heard of anyone having that error when they have created a new valid token and set the new token using the method described in the instructions in RStudio. I assume you had the token working correctly as recently as yesterday given you created rebeccamayer1Diversity? If you find it impossible to connect RStudio to GitHub though, then I'll mark what you've given me a link to in your submission assuming I can download it. It is obviously unfortunate you have a totally new error a day before the submission deadline, having not had any problems in the first 5 weeks after the project was set. In the meanwhile I've extended to submission deadline by a day in case you can fix it tomorrow.

**Reply by RebeccaMayer** [reply]
there is no error when I follow the instructions but the push button is still disabled after I commit the files, thank you for the extension!

**Reply by richardreeve** [reply]
If there are no errors after you have a committed a change to the repository and there is still no push button then it seems that it is not connected to github, so you should do:


```R
usethis::use_github(organisation = "SBOHVM", private = TRUE)
```

if that errors or anyway doesn’t light up the push button it seems very likely you did something wrong in the initial instructions because RStudio didn’t have valid github credentials. I would Google for how to push/upload a new local git repo to GitHub.

**Reply by richardreeve** [reply]
Having downloaded what you sent me, I would observe that you haven't committed anything to git, so there may be nothing to push to github. Just a thought.

**Comment by RebeccaMayer** [comment, marked as answer]
Hi @richardreeve, I have managed to push everything to GitHub. Still not sure what the issue was, I had to create a new repository without the readme.md file, then add it through my terminal in R, which reactivated the push and pull buttons. I have updated the link in moodle, thanks again for all your help!
